### PR TITLE
Small improvement of showKind get

### PR DESCRIPTION
**Pull request by niuzhenguo**
**What this PR does / why we need it**:
Given the complexity of determining whether to show resource with kind, this PR will be a small improvement when with `--show-kind=true`. There's no need to waste time to run the complex logic if we already decided to show resource with kind.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/resource/get.go [hunk 1]
```diff
@@ -303,8 +303,10 @@ func (options *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []str
 
 	showKind := options.ShowKind
 	// TODO: abstract more cleanly
-	if resource.MultipleTypesRequested(args) || cmdutil.MustPrintWithKinds(objs, infos, sorter) {
-		showKind = true
+	if !showKind {
```
Comment (sttts) [inline pkg/kubectl/cmd/resource/get.go]: what does this change?

Comment (sttts) [reply pkg/kubectl/cmd/resource/get.go]: What is the complexity of the checks below? Is there any IO involved, e.g. network?

Comment (niuzhenguo) [reply pkg/kubectl/cmd/resource/get.go]: there's no IO involved, just some logic improvement. If we already set ShowKind, there's absolutely no need to check MultipleTypesRequested or MustPrintWithKinds to decide we need to show kind.

**Comment by niuzhenguo** [comment]
cc @dixudx @mengqiy

File: pkg/kubectl/cmd/resource/get.go (hunk 1 above)
Comment (dixudx) [reply pkg/kubectl/cmd/resource/get.go]: If so, why not changing directly to
```
showKind := options.ShowKind || resource.MultipleTypesRequested(args) || cmdutil.MustPrintWithKinds(objs, infos, sorter)
```

Comment (niuzhenguo) [reply pkg/kubectl/cmd/resource/get.go]: seems more elegant, will update soon, thanks!

**Comment by niuzhenguo** [comment]
@dixudx Done

**Pull request merged**